Tests/Tests/CampaignRepositorySpec.swift

```diff
@@ -273,6 +287,12 @@ class CampaignRepositorySpec: QuickSpec {
                     campaignRepository.optOutCampaign(testCampaign)
                     expect(userCache?.campaignData?.first?.isOptedOut).to(beFalse())
                 }
+
+                it("optout campaign with invalid campaign") {
```

Comment:
Please move this test to context("when optOutCampaign is called") context below and create an additional context("when provided campaign is invalid") context and rename the test to "will not mark campaign as opted out".

Reply:
updated